### PR TITLE
Show metadata form when purchasing a key

### PR DESCRIPTION
**Pull request by cnasc**
# Description

This PR introduces a metadata collection flow to key purchases.

In essence, when `metadataInputs` are specified in the paywall config the checkout modal will now prompt for the user to fill out the fields before the key purchase occurs.

There is probably more that can be done to refine this particular flow, but it will take bigger shifts in the paywall to get there.

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by cnasc** [comment]
image

The current code works to a degree. It will show the metadata form when inputs are specified, metamask will pop up to do the signature, then the key purchase will initiate (result of successful run shown above).

However, the way we show the wallet check message clips the checkout component out of the tree, which interrupts all the listeners. If we show the wallet check when getting the signature, it breaks the flow. So before this is ready for prime time we'll have to figure out a better way to work with that.

**Comment by cnasc** [comment]
As far as I can tell, the resolution here is going to be moving the wallet check message down a layer into the `<Checkout>` component.

**Comment by cnasc** [comment]
> As far as I can tell, the resolution here is going to be moving the wallet check message down a layer into the `<Checkout>` component.

This did not resolve the issue

**Comment by cnasc** [comment]
Right now there is almost no unit-test coverage of the checkout flow, everything is deferred to the integration tests. LMK if integration tests should be added as part of this PR or as a separate follow-up.

I'm happy to demo the paywall with metadata collection in localdev if that will ease review.

**Comment by julien51** [comment]
I think it would be good to write integration tests but I worry this is going to end up quite intense for logic which we'll soon move to the unlock-app so let's write some when we get there.

File: paywall/src/components/content/NewDemoContent.tsx [hunk 1]
```diff
@@ -49,6 +49,8 @@ export default function NewDemoContent() {
   const names: string[] = url.searchParams.getAll('name')
   const unlockUserAccounts: boolean =
     url.searchParams.get('unlockUserAccounts') === 'true'
+  const useMetadataInputs: boolean =
+    url.searchParams.get('metadataInputs') === 'true'
```
Comment (julien51) [inline paywall/src/components/content/NewDemoContent.tsx]: 👍

File: paywall/src/components/interface/MetadataForm.tsx [hunk 2]
```diff
@@ -26,7 +37,10 @@ export const MetadataForm = ({ fields, onSubmit }: Props) => {
           </Label>
         ))}
       </FieldWrapper>
-      <ActionButton type="submit">Continue</ActionButton>
+      {submittedForm && (
```
Comment (julien51) [inline paywall/src/components/interface/MetadataForm.tsx]: 👍

**Pull request merged**